### ISSUE TITLE
Hello Tensorflow.

### ISSUE THREAD

**Issue by HuangJiaLian** [HuangJiaLian/DNN-ODE-Solver#4]
Before solving ODEs with Tensorflow, the basics of TensorFlow should be mastered.  You are going to classify handwritten digits by training a deep neural network(DNN) with Tensorflow. 

The [MNIST](http://yann.lecun.com/exdb/mnist/) database of handwritten digits, available from this page, has a training set of 60,000 examples, and a test set of 10,000 examples.

<p align='center'>
<img src='https://raw.githubusercontent.com/HuangJiaLian/DataBase0/master/uPic/yzf3kx.jpg' width='50%'>
</p>

## Todo
- [ ] Train a worked DNN to classify handwritten digits using TensorFlow.
- [ ] Answer the following questions. Show data if possible.
- [ ] Build another DNN to classify the officemates. (Build your first face recognition application. 😎)

**Basic Questions**:
- How to build DNN using Tensorflow?
- What is activation functions, how to choose them?
- What loss functions mean, and how to write a loss function for this handwritten digits classification problem? 
- How to train a DNN? 
- What is the learning rate, and what are the effects of different learning rates?
- What is a mini-batch, and what are the effects of different sizes of mini-batch? 
- What is training data,  training set? 
- What is the purpose of the testing set? 
- What is overfitting, and how to avoid overfitting? 
- When to stop training? 
- How to save trained DNN? 
- How to use a trained DNN? 

**Other questions** 
- How to write a loss function by yourself?
- How to obtain the derivative of the output of DNN? 

## Tips
Links may be useful.

[Machine Learning Foundations: Ep #1 - What is ML?](https://www.youtube.com/watch?v=_Z9TRANg4c0)
[Machine Learning Foundations: Ep #2 - First steps in computer vision](https://youtu.be/j-35y1M9rRU)

@Fang-max  @974918051 @yychenleslie
